[Matthew Liu Zhen Jie] iP

matthew-liu-zhenjie opened the pull request.

matthew-liu-zhenjie committed:
Update Duke.java

matthew-liu-zhenjie committed:
Add user input and echo functionality

matthew-liu-zhenjie committed:
Added list to store text and display list

matthew-liu-zhenjie committed:
Add Task class, update Duke to mark tasks as done

matthew-liu-zhenjie committed:
Update Duke.java

Extract marking of tasks in handleRequest into markTasks method
Edited handleRequest method to print numbers for tasks printed when user requests "list".
Renamed method dashSeperator to printSeparator
Renamed method sayFeedback to printTaskDetails for better clarity
Renamed method sayEcho to printEcho
Renamed method sayGreeting to printGreeting
Renamed method sayGoodbye to printGoodbye

matthew-liu-zhenjie committed:
Add TaskManager

Refactored methods involving taskList into TaskManager class

matthew-liu-zhenjie committed:
Add ToDo, Deadline and Event classes. Update Duke, Task Manager and Task classes

matthew-liu-zhenjie committed:
Update Duke, Task and TaskManager

Added a method to print commands if none of the valid commands are entered.
Code was edited to comply with coding standards

matthew-liu-zhenjie committed:
Add test case

Ensure that output is consistent with expectations

khooyourun commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,50 @@
-        System.out.println("Hello from\n" + logo);
+        //System.out.println("Hello from\n" + logo);
+        printGreeting();
+        boolean byeFlag = false;
```
Better to name Boolean variables in a way that sounds like a boolean

khooyourun commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,50 @@
+        printSeparator();
+        System.out.println("added: " + input);
+        printSeparator();
+    }
```
I like the naming of methods here. It makes it clear what each method does.

khooyourun commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,107 @@
+        } else{
+            printHelp();
+        }
+    }
```
I like the usage of else if statements instead of nested if statements. Makes it easy to understand.

khooyourun commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,107 @@
+        taskList[taskIndex-1].markAsUndone();
+    }
+    public void markTask(String input) {
+        int whitespaceIndex = input.indexOf(" ");
```
Preferably name whitespaceIndex as whiteSpaceIndex with a capital S (camelCase).

matthew-liu-zhenjie replied:
Hey khooyourun, thanks for the comment. However, whitespace is a single word

khooyourun commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,107 @@
+    }
+
+    private void generateToDo(String input) {
+        String job = input.substring(5);
```
What does the 5 in input.substring(5) represent? Same question for the other input.substring()

matthew-liu-zhenjie replied:
Noted, I'll comment as required for clarity. This creates a substring from input[5] onwards.

khooyourun reviewed (commented):
Overall the code is easy to read as it uses well named modules and variables.

matthew-liu-zhenjie committed:
Add duke.command and duke.task packages. Refactor files into packages

Moved files into packages to make code more organized

matthew-liu-zhenjie committed:
Update TaskManager. Add EmptyTaskException and IllegalCommandException.

Changed implementation of handleCommand method to seperate the command keyword for better exception handling.

Added EmptyTaskException and IllegalCommandException to catch erroneous user input.

matthew-liu-zhenjie committed:
Update input

matthew-liu-zhenjie committed:
Merge branch 'master' into branch-Level-5-and-A-Packages

matthew-liu-zhenjie committed:
Add delete method

Enables user to delete tasks in the task list. task list implementation was changed from static array to array list to enable easy deletion.

matthew-liu-zhenjie committed:
Add SaveManager Package, update TaskManager class

matthew-liu-zhenjie committed:
Merge commit '8a608b46920385e01ccc748a29aa08a7fd20512e'

# Conflicts:
#	src/main/java/duke/command/TaskManager.java

matthew-liu-zhenjie committed:
Update Duke and TaskManager, add UI class

matthew-liu-zhenjie committed:
Refactor TaskManager to TaskList

matthew-liu-zhenjie committed:
Refactor SaveState and LoadState to Storage class.

matthew-liu-zhenjie committed:
Add Parser. Update Ui class.

matthew-liu-zhenjie committed:
Refactor Duke class from default branch to duke package.

matthew-liu-zhenjie committed:
Add find feature

matthew-liu-zhenjie committed:
Add JavaDocs documentation.

matthew-liu-zhenjie committed:
Merge branch 'master' into level-9_Find

# Conflicts:
#	src/main/java/duke/parser/Parser.java

matthew-liu-zhenjie committed:
Fix serialization/deserialization bug in Storage.java

matthew-liu-zhenjie committed:
Fix bug where /from and /to are swapped in Event command results in Unexpected behavior

matthew-liu-zhenjie committed:
Merge branch 'level-9_Find'

matthew-liu-zhenjie committed:
Update README.md

Add About and Features

matthew-liu-zhenjie committed:
Update README.md

matthew-liu-zhenjie committed:
Update README.md

matthew-liu-zhenjie committed:
Update README.md

matthew-liu-zhenjie committed:
Update README.md

matthew-liu-zhenjie committed:
Fix bug in Ui where find index displayed is not offset.

matthew-liu-zhenjie committed:
Merge remote-tracking branch 'origin/master'

matthew-liu-zhenjie committed:
Fix bug where "/by" is not checked when generating deadline.

matthew-liu-zhenjie committed:
Update README.md

matthew-liu-zhenjie committed:
Update Ui printHelp method.

matthew-liu-zhenjie committed:
Merge remote-tracking branch 'origin/master'
